components/ws-proxy/pkg/proxy/infoprovider.go

```diff
@@ -400,7 +402,24 @@ func (c *workspaceInfoCache) Insert(info *WorkspaceInfo) {
 }
 
 func (c *workspaceInfoCache) doInsert(info *WorkspaceInfo) {
-	c.infos[info.WorkspaceID] = info
+	existingInfos := c.infos[info.WorkspaceID]
+	// Only insert if this is a new instance
+	mustAppend := true
+	for i, existingInfo := range existingInfos {
+		if existingInfo.InstanceID == info.InstanceID {
+			c.infos[info.WorkspaceID][i] = info
+			log.WithField("info", info).Debug("ignoring insert of existing instance for workspace")
```

Comment:
nit: this log statement will be very verbose in prod

Reply:
It is debug, but I'd left it for testing purposes - removed